Failure looks similar to #85082 but the pattern there is different.

```
CSC : error CS8034: Unable to load Analyzer assembly /__w/1/s/.dotnet/sdk/8.0.100-preview.3.23178.7/Sdks/Microsoft.NET.Sdk/analyzers/Microsoft.CodeAnalysis.CSharp.NetAnalyzers.dll : The file '/tmp/VBCSCompiler/AnalyzerAssemblyLoader/a6fe1c15dfaa4493b9e3eda990245fe8/30' already exists. [/__w/1/s/src/tests/Common/XUnitWrapperGenerator/XUnitWrapperGenerator.csproj] [/__w/1/s/src/tests/build.proj]
```

--- a/src/coreclr/jit/fgdiagnostic.cpp
+++ b/src/coreclr/jit/fgdiagnostic.cpp
@@ -4205,11 +4205,25 @@ public:
 
     void CheckPhis(BasicBlock* block)
     {
+        Statement* nonPhiStmt = nullptr;
         for (Statement* const stmt : block->Statements())
         {
+            // All PhiDefs should appear before any other statements
+            //
             if (!stmt->IsPhiDefnStmt())
             {
-                break;
+                if (nonPhiStmt == nullptr)
+                {
+                    nonPhiStmt = stmt;
+                }
+                continue;
+            }
+
+            if (nonPhiStmt != nullptr)
+            {
+                SetHasErrors();
+                JITDUMP("[error] " FMT_BB " PhiDef " FMT_STMT " appears after non-PhiDef " FMT_STMT "\n", block->bbNum,
+                        stmt->GetID(), nonPhiStmt->GetID());
             }
 
             GenTree* const phiDefNode = stmt->GetRootNode();

